### PR TITLE
tidb: rename products

### PR DESCRIPTION
**Pull request by qiancai**
### What is changed, added or deleted? (Required)

- Remove the use of "on-premises" and similar ones
- Update "TiDB Cloud Serverless Tier" and similar ones to "TiDB Serverless"
- Update affected anchor links

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [x] v7.0 (TiDB 7.0 versions)
- [x] v6.6 (TiDB 6.6 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/docs/pull/13668

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 1]
```diff
@@ -1,30 +1,30 @@
 ---
-title: 使用 TiDB Cloud (Serverless Tier) 构建 TiDB 集群
-summary: 使用 TiDB Cloud (Serverless Tier) 构建 TiDB 集群，并连接 TiDB Cloud 集群。
+title: 使用 TiDB Serverless 构建 TiDB 集群
+summary: 使用 TiDB Serverless 构建 TiDB 集群，并连接 TiDB Cloud 集群。
```
Comment (ran-huang) [inline develop/dev-guide-build-cluster-in-cloud.md:3]:
```suggestion
summary: 使用 TiDB Serverless 构建 TiDB 集群，并连接 TiDB Serverless 集群。
```

File: tiflash-performance-tuning-methods.md [hunk 2]
```diff
@@ -98,7 +98,7 @@ summary: 本文介绍了 Performance Overview 面板中 TiFlash 部分，帮助
 
 你可以通过 `(Read flow + Write flow) ÷ 总的 Write Throughput By Instance` 计算出整个 TiFlash 集群的写放大倍数。
 
-示例 1 ：[CH-benCHmark 负载](/benchmark/benchmark-tidb-using-ch.md) OP 环境 Raft 和 IO 指标
+示例 1 ：[CH-benCHmark 负载](/benchmark/benchmark-tidb-using-ch.md) 本地部署境 Raft 和 IO 指标
```
Comment (ran-huang) [inline tiflash-performance-tuning-methods.md:101]:
```suggestion
示例 1 ：[CH-benCHmark 负载](/benchmark/benchmark-tidb-using-ch.md) 本地部署环境 Raft 和 IO 指标
```

File: develop/dev-guide-tidb-crud-sql.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ aliases: ['/zh/tidb/dev/tidb-crud-sql']
 
 ## 在开始之前
 
-请确保你已经连接到 TiDB 集群，若未连接，请参考[使用 TiDB Cloud (Serverless Tier) 构建 TiDB 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)来创建一个 Serverless Tier 集群。
+请确保你已经连接到 TiDB 集群，若未连接，请参考[使用 TiDB Serverless 构建 TiDB 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)来创建一个 TiDB Serverless 集群。
```
Comment (lilin90) [inline develop/dev-guide-tidb-crud-sql.md:13]:
```suggestion
请确保你已经连接到 TiDB 集群，若未连接，请参考[使用 TiDB Serverless 构建 TiDB 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)来创建一个 TiDB Serverless 集群。
```

File: develop/dev-guide-build-cluster-in-cloud.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ aliases: ['/zh/tidb/dev/build-cluster-in-cloud']
 
     > **注意：**
     >
-    > 在连接到 [Serverless Tier](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier) 集群时，你需要给用户名加上前缀并使用单引号包裹用户名。你可以在 [TiDB Cloud 用户名前缀](https://docs.pingcap.com/tidbcloud/select-cluster-tier#user-name-prefix) 中获得更多信息。
+    > 在连接到 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier) 集群时，你需要给用户名加上前缀并使用单引号包裹用户名。你可以在 [TiDB Cloud 用户名前缀](https://docs.pingcap.com/tidbcloud/select-cluster-tier#user-name-prefix) 中获得更多信息。
```
Comment (lilin90) [inline develop/dev-guide-build-cluster-in-cloud.md:37]: As in https://github.com/pingcap/docs/pull/13668/files#diff-ec4e6ff3d33b51901406c57b6149d05efec6cc6650f1a04b042b904a92518853
```suggestion
    > 在连接到 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-serverless-beta) 集群时，你需要给用户名加上前缀并使用单引号包裹用户名。你可以在 [TiDB Cloud 用户名前缀](https://docs.pingcap.com/tidbcloud/select-cluster-tier#user-name-prefix) 中获得更多信息。
```

File: develop/dev-guide-sample-application-django.md [hunk 5]
```diff
@@ -29,7 +29,7 @@ summary: 给出一个 Django 构建 TiDB 应用程序示例。
 
 <div label="TiDB Cloud" value="serverless-cluster">
 
-[创建 Serverless Tier 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
+[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
```
Comment (lilin90) [inline develop/dev-guide-sample-application-django.md:32]:
```suggestion
[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
```

File: develop/dev-guide-sample-application-golang.md [hunk 6]
```diff
@@ -22,7 +22,7 @@ summary: 给出一个 TiDB 和 Golang 的简单 CRUD 应用程序示例。
 
 <div label="TiDB Cloud" value="serverless-cluster">
 
-[创建 Serverless Tier 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
+[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
```
Comment (lilin90) [inline develop/dev-guide-sample-application-golang.md:25]:
```suggestion
[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
```

File: develop/dev-guide-sample-application-java.md [hunk 7]
```diff
@@ -32,7 +32,7 @@ aliases: ['/zh/tidb/dev/sample-application-java']
 
 <div label="TiDB Cloud" value="serverless-cluster">
 
-[创建 Serverless Tier 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
+[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
```
Comment (lilin90) [inline develop/dev-guide-sample-application-java.md:35]:
```suggestion
[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
```

File: develop/dev-guide-sample-application-python.md [hunk 8]
```diff
@@ -22,7 +22,7 @@ summary: 给出一个 TiDB 和 Python 的简单 CRUD 应用程序示例。
 
 <div label="TiDB Cloud" value="serverless-cluster">
 
-[创建 Serverless Tier 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
+[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
```
Comment (lilin90) [inline develop/dev-guide-sample-application-python.md:25]:
```suggestion
[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
```

File: develop/dev-guide-sample-application-spring-boot.md [hunk 9]
```diff
@@ -31,7 +31,7 @@ aliases: ['/zh/tidb/dev/sample-application-spring-boot']
 
 <div label="TiDB Cloud" value="serverless-cluster">
 
-[创建 Serverless Tier 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
+[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-serverless-tier-集群)。
```
Comment (lilin90) [inline develop/dev-guide-sample-application-spring-boot.md:34]:
```suggestion
[创建 TiDB Serverless 集群](/develop/dev-guide-build-cluster-in-cloud.md#第-1-步创建-tidb-serverless-集群)。
```

File: releases/release-7.0.0.md [hunk 10]
```diff
@@ -248,7 +248,7 @@ TiDB 版本：7.0.0
 * [DBeaver](https://dbeaver.io/) v23.0.1 默认支持 TiDB [#17396](https://github.com/dbeaver/dbeaver/issues/17396) @[Icemap](https://github.com/Icemap)
 
     - 提供独立的 TiDB 模块、Icon 和标识。
-    - 默认配置支持 [TiDB Cloud Serverless Tier](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta)，你可以更方便地连接 Serverless Tier。
+    - 默认配置支持 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta)，你可以更方便地连接 TiDB Serverless。
```
Comment (lilin90) [inline releases/release-7.0.0.md:251]:
```suggestion
    - 默认配置支持 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-serverless-beta)，你可以更方便地连接 TiDB Serverless。
```

File: time-to-live.md [hunk 11]
```diff
@@ -238,7 +238,7 @@ TTL 功能能够与 TiDB 的迁移、备份、恢复工具一同使用。
 * 具有 TTL 属性的表不支持作为外键约束的主表被其他表引用。
 * 不保证所有过期数据立即被删除，过期数据被删除的时间取决于后台清理任务的调度周期和调度窗口。
 * 对于使用[聚簇索引](/clustered-indexes.md)的表，如果主键的类型不是整数类型或二进制字符串类型，TTL 任务将无法被拆分成多个子任务。这将导致 TTL 任务只能在一个 TiDB 节点上按顺序执行。如果表中的数据量较大，TTL 任务的执行可能会变得缓慢。
-* TTL 无法在 TiDB Cloud [Serverless Tier](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta) 集群上使用。
+* TTL 无法在 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#serverless-tier-beta) 集群上使用。
```
Comment (lilin90) [inline time-to-live.md:241]:
```suggestion
* TTL 无法在 [TiDB Serverless](https://docs.pingcap.com/tidbcloud/select-cluster-tier#tidb-serverless-beta) 集群上使用。
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Review by ran-huang — APPROVED** [review]

**Comment by Oreoxmt** [comment]
@qiancai @ran-huang https://github.com/pingcap/docs-cn/pull/14062/commits/768f4be719244c1c90309a23271bfb1ee5e7a5c6 PTAL

**Pull request merged**